packages/api/internal/orchestrator/nodemanager/metadata.go

```diff
@@ -33,44 +33,49 @@ func (n *Node) Metadata() NodeMetadata {
-func (n *Node) getClientMetadata() metadata.MD {
-	return metadata.New(map[string]string{consts.EdgeRpcServiceInstanceIDHeader: n.Metadata().ServiceInstanceID})
-}
+func (n *Node) GetSandboxCreateCtx(ctx context.Context, req *orchestrator.SandboxCreateRequest) (*clusters.GRPCClient, context.Context) {
+	md := metadata.MD{}
 
-func (n *Node) GetSandboxCreateCtx(ctx context.Context, req *orchestrator.SandboxCreateRequest) context.Context {
-	// Skip local cluster. It should be okay to send it here, but we don't want to do it until we explicitly support it.
-	if n.IsNomadManaged() {
-		return ctx
-	}
+	if !n.IsLocal() {
+		md = edge.SerializeSandboxCatalogCreateEvent(
+			edge.SandboxCatalogCreateEvent{
+				SandboxID:               req.GetSandbox().GetSandboxId(),
+				SandboxMaxLengthInHours: req.GetSandbox().GetMaxSandboxLength(),
+				SandboxStartTime:        req.GetStartTime().AsTime(),
 
-	md := edge.SerializeSandboxCatalogCreateEvent(
-		edge.SandboxCatalogCreateEvent{
-			SandboxID:               req.GetSandbox().GetSandboxId(),
-			SandboxMaxLengthInHours: req.GetSandbox().GetMaxSandboxLength(),
-			SandboxStartTime:        req.GetStartTime().AsTime(),
+				ExecutionID:    req.GetSandbox().GetExecutionId(),
+				OrchestratorID: n.Metadata().ServiceInstanceID,
+			},
+		)
+	}
 
-			ExecutionID:    req.GetSandbox().GetExecutionId(),
-			OrchestratorID: n.Metadata().ServiceInstanceID,
-		},
-	)
+	// return n.connection, metadata.AppendToOutgoingContext(ctx, args...)
 
-	return metadata.NewOutgoingContext(ctx, metadata.Join(n.getClientMetadata(), md))
+	// Merge medata from client connection (auth, routing with service instance id) and event metadata.
+	return n.connection, appendMetadataCtx(ctx, md)
 }
 
-func (n *Node) GetSandboxDeleteCtx(ctx context.Context, sandboxID string, executionID string) context.Context {
-	// Skip local cluster. It should be okay to send it here, but we don't want to do it until we explicitly support it.
-	if n.IsNomadManaged() {
-		return ctx
+func (n *Node) GetSandboxDeleteCtx(ctx context.Context, sandboxID string, executionID string) (*clusters.GRPCClient, context.Context) {
+	md := metadata.MD{}
+
+	if !n.IsLocal() {
+		md = edge.SerializeSandboxCatalogDeleteEvent(
+			edge.SandboxCatalogDeleteEvent{
+				SandboxID:   sandboxID,
+				ExecutionID: executionID,
+			},
+		)
 	}
 
-	md := edge.SerializeSandboxCatalogDeleteEvent(
-		edge.SandboxCatalogDeleteEvent{
-			SandboxID:   sandboxID,
-			ExecutionID: executionID,
-		},
-	)
+	// Merge medata from client connection (auth, routing with service instance id) and event metadata.
+	return n.connection, appendMetadataCtx(ctx, md)
+}
+
+func appendMetadataCtx(ctx context.Context, md metadata.MD) context.Context {
+	args := make([]string, 0, len(md)*2)
+	for k, v := range md {
+		args = append(args, k, v[0])
+	}
 
-	return metadata.NewOutgoingContext(ctx, metadata.Join(n.getClientMetadata(), md))
+	return metadata.AppendToOutgoingContext(ctx, args...)
```

Comment:
We are now using `AppendToOutgoingContext` method that only appends metadata, the previous `NewOutgoingContext` cleared metadata set during connection creation (auth and service instance id).